### PR TITLE
Use model constructors when creating history instances

### PR DESCRIPTION
**Pull request by allait**
`create_history` used to initialise an empty model instance and assign attributes to it one by one. Assigned attributes are collected by iterating over internal state of the modified record and stored in a dictionary, so the order in which they are set on the history instance is undefined.

This works fine for most model instances, but recent changes to the Template models require `template_type` to be set before `reply_to`, otherwise Template doesn't know which column to use to persist the `reply_to` value.

This means that we need to order the attributes when they're being assigned. We achieved this by defining a constructor for Template objects that assigns `template_type` first.

In order to make it work with `create_history` we replace individual `setattr` calls with a single call to the Template constructor which passes the entire data dictionary as keyword arguments.

This works because all persisted attributes are defined as columns and SQLAlchemy accepts all column values as constructor arguments.

### CODE REVIEW

File: tests/app/template/test_rest.py [hunk 1]
```diff
@@ -616,6 +616,37 @@ def test_update_set_process_type_on_template(client, sample_template):
...
+    auth_header = create_authorization_header()
+
+    response = client.post(
+        '/service/{}/template'.format(service.id),
+        headers=[('Content-Type', 'application/json'), auth_header],
+        data=data
+    )
+    assert response.status_code == 201, response.get_data(as_text=True)
+
+    json_resp = json.loads(response.get_data(as_text=True))
```
Comment (leohemsted) [inline tests/app/template/test_rest.py]: if you use the `admin_request` fixture, it does a lot of the gubbins around setting up a request - json handling, headers, URL creation, etc.

```python
json_resp = admin_request.post('template.template', service_id=service.id, _data=data, _expected_status=201)
```

File: tests/app/template/test_rest.py [hunk 2]
```diff
@@ -616,6 +616,37 @@ def test_update_set_process_type_on_template(client, sample_template):
...
+    assert json_resp['data']['template_type'] == 'letter'
+    assert json_resp['data']['reply_to'] == str(letter_contact.id)
+
+    template = Template.query.get(json_resp['data']['id'])
+    from app.schemas import template_schema
```
Comment (leohemsted) [inline tests/app/template/test_rest.py]: can this go up the top with the rest of the imports?

Comment (allait) [reply tests/app/template/test_rest.py]: I originally copied it from the test above and it looks like all tests delay this import like this.
It turns out 59 tests fail if this is moved to module level. My guess is marshmallow stores some db/session reference somewhere and if that happens before the session fixtures are applied (eg during test module import) it gets stuck with a `DummySession`.

I'm not really sure what the implications of this are.

Comment (leohemsted) [reply tests/app/template/test_rest.py]: 😩

**Pull request merged**